api/migrations/20180319105352_add-expenses.js

```diff
@@ -0,0 +1,25 @@
+exports.up = async knex => {
+  await knex.schema.createTable('expenses', table => {
+    table.increments('id');
+    table.string('name', 64).unique();
+  });
+
+  await knex.schema.createTable('activity_expenses', table => {
+    table.string('year', 4);
+    table.decimal('amount', 10, 2);
+    table.text('description');
+
+    table.integer('expense_id');
+    table.integer('activity_id');
+
+    table.index(['expense_id', 'activity_id']);
+
+    table.foreign('expense_id').references('expenses.id');
+    table.foreign('activity_id').references('activities.id');
+  });
+};
```

Comment:
I'm curious why there are two tables here.  The `expenses` table only adds a name.  Is the idea that a single name may be shared by multiple expenses (e.g., "paperclips" for 2018, 2019, and 2020)?

Reply:
yes, and same name could be shared by multiple activities or apds, too, i think; this also paves the way for more intricate expense hierarchies, like expense subcategories